internal/newnode/newnode.go

```diff
@@ -72,32 +80,34 @@ func (node *NewNode) requestBeaconChain(BCPeer p2p.Peer) (err error) {
 	if err != nil {
 		node.log.Error("Could not Marshall public key into binary")
 	}
-	p := p2p.Peer{IP: node.Self.IP, Port: node.Self.Port}
-	nodeInfo := &bcconn.NodeInfo{Self: p, PubK: pubk}
+	nodeInfo := &proto_node.Info{IP: node.Self.IP, Port: node.Self.Port, PubKey: pubk, PeerID: node.host.GetID()}
 	msg := bcconn.SerializeNodeInfo(nodeInfo)
 	msgToSend := proto_identity.ConstructIdentityMessage(proto_identity.Register, msg)
 	gotShardInfo := false
-	timeout := time.After(300 * time.Second)
-	tick := time.Tick(5 * time.Second)
+	timeout := time.After(120 * time.Second)
+	tick := time.Tick(3 * time.Second)
 checkLoop:
 	for {
 		select {
 		case <-timeout:
 			gotShardInfo = false
 			break checkLoop
 		case <-tick:
-			if node.SetInfo {
-				gotShardInfo = true
-				break checkLoop
-			} else {
+			select {
+			case setinfo := <-node.SetInfo:
+				if setinfo {
+					gotShardInfo = true
+					break checkLoop
+				}
+			default:
 				host.SendMessage(node.host, BCPeer, msgToSend, nil)
 			}
 		}
 	}
 	if !gotShardInfo {
 		err = errors.New("could not create connection")
-		node.log.Crit("Could not get sharding info after 5 minutes")
-		os.Exit(1)
+		node.log.Crit("Could not get sharding info after 2 minutes")
+		os.Exit(10)
```

Comment:
can you elaborate this? 10 or 1?

Reply:
make it 10 to make it easier to identify the failure reason in return code of the go test.  Again, newnode will be obsolete soon.